qutebrowser/browser/shared.py

```diff
@@ -355,23 +355,60 @@ def choose_file(multiple: bool) -> List[str]:
         A list of selected file paths, or empty list if no file is selected.
         If multiple is False, the return value will have at most 1 item.
     """
-    handle = tempfile.NamedTemporaryFile(prefix='qutebrowser-fileselect-', delete=False)
-    handle.close()
-    tmpfilename = handle.name
-    with utils.cleanup_file(tmpfilename):
-        if multiple:
-            command = config.val.fileselect.multiple_files.command
-        else:
-            command = config.val.fileselect.single_file.command
+    if multiple:
+        command = config.val.fileselect.multiple_files.command
+    else:
+        command = config.val.fileselect.single_file.command
+    use_tmp_file = '{}' in command
+    if use_tmp_file:
+        handle = tempfile.NamedTemporaryFile(
+            prefix='qutebrowser-fileselect-',
+            delete=False,
+        )
+        handle.close()
+        tmpfilename = handle.name
+        command = (
+            command[:1] +
+            [arg.replace('{}', tmpfilename) for arg in command[1:]]
+        )
+        with utils.cleanup_file(tmpfilename):
+            return execute_fileselect_command(
+                command=command,
+                multiple=multiple,
+                tmpfilename=tmpfilename,
+            )
+    else:
+        return execute_fileselect_command(
+            command=command,
+            multiple=multiple,
+        )
 
-        proc = guiprocess.GUIProcess(what='choose-file')
-        proc.start(command[0],
-                   [arg.replace('{}', tmpfilename) for arg in command[1:]])
 
-        loop = qtutils.EventLoop()
-        proc.finished.connect(lambda _code, _status: loop.exit())
-        loop.exec()
+def execute_fileselect_command(
```

Comment:
```suggestion
def _execute_fileselect_command(
```

Since it's a "private" utility function, not used outside of this module.

Reply:
Fixed